pkg/activitypub/resthandler/activityhandler.go

```diff
@@ -292,23 +301,31 @@ func (h *Activity) handle(w http.ResponseWriter, req *http.Request) {
 		if errors.Is(err, spi.ErrNotFound) {
 			logger.Debugf("[%s] Activity ID not found [%s]", h.endpoint, activityIRI)
 
-			h.writeResponse(w, http.StatusNotFound, nil)
+			h.writeResponse(w, http.StatusNotFound, []byte("Not Found.\n"))
 
 			return
 		}
 
 		logger.Errorf("[%s] Unable to retrieve activity [%s]: %s", h.endpoint, activityIRI, err)
 
-		h.writeResponse(w, http.StatusInternalServerError, nil)
+		h.writeResponse(w, http.StatusInternalServerError, []byte("Internal Server Error.\n"))
 
 		return
 	}
 
+	if !authorized {
+		if !activity.To().Contains(vocab.PublicIRI) {
+			h.writeResponse(w, http.StatusUnauthorized, []byte("Unauthorised.\n"))
```

Comment:
Better to use consistent spelling on "unauthorized" - (you have a `s` here)

Reply:
I have another PR coming. I'll change it then.